### PR TITLE
Add back support for changing the kafka library

### PR DESCRIPTION
**Pull request by GaryWilber**
Support for this was removed in https://github.com/microsoft/FluidFramework/pull/5120
Adding back the ability to change the kafka lib via the `kafka:lib:name` setting.

Other change included - remove myself from odsp-driver prs as I do not actively work on it

### CODE REVIEW

File: server/routerlicious/packages/routerlicious/src/kafka-service/resourcesFactory.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import { KafkaResources, KafkaResourcesFactory } from "@fluidframework/server-services-ordering-kafkanode";
+import { RdkafkaResourcesFactory } from "@fluidframework/server-services-ordering-rdkafka";
+import { IResourcesFactory } from "@fluidframework/server-services-utils";
+import { Provider } from "nconf";
+
+/**
+ * A generic kafka resources factory that picks rdkafka / kafka-node based on the config
+ */
+export class OrderingResourcesFactory implements IResourcesFactory<KafkaResources> {
```
Comment (hedasilv) [inline server/routerlicious/packages/routerlicious/src/kafka-service/resourcesFactory.ts:14]: The RdKafkaResources factory create() method returns a Promise<RdkafkaResources>.
Should create() from this new OrderingResourceFactory return a Promise<IKafkaResources> instead?

In other words, is the KafkaResources type compatible with RdkafkaResources?

File: server/routerlicious/packages/routerlicious/src/kafka-service/index.ts [hunk 2]
```diff
@@ -4,10 +4,10 @@
...
 
 import * as path from "path";
-import { KafkaResourcesFactory } from "@fluidframework/server-services-ordering-kafkanode";
 
 import { execute } from "./command";
+import { OrderingResourcesFactory } from "./resourcesFactory";
 
 execute(
-    (name: string, lambda: string) => new KafkaResourcesFactory(name, lambda),
+    (name: string, lambda: string) => new OrderingResourcesFactory(name, lambda),
```
Comment (hedasilv) [inline server/routerlicious/packages/routerlicious/src/kafka-service/index.ts:12]: Thanks Gary!

File: server/routerlicious/packages/routerlicious/src/kafka-service/resourcesFactory.ts (hunk 1 above)
Comment (GaryWilber) [reply server/routerlicious/packages/routerlicious/src/kafka-service/resourcesFactory.ts:14]: `RdkafkaResources` is compatible with `KafkaResources`. They are actually the same interface / class but with different names. Ideally they wouldn't be separate and they should share the name resources class. However it was hard to find a shared place in the codebase to put that class. So for now they are separate.

**Review by hedasilv — APPROVED** [review]

**Comment by znewton** [comment]
@GaryWilber could this new OrderingResourceFactory be moved to @fluidframework/server-routerlicious-base (FFX/server/routerlicious/packages/routerlicious-base)? This would make it consumable from custom R11S implementations, similar to the Alfred and Riddler modules exported from r11s-base.

**Review by hedasilv — APPROVED** [review]

**Review by tanviraumi — APPROVED** [review]

**Pull request merged**